src/lib/pages/proposal/whitelist/index.tsx

```diff
@@ -0,0 +1,428 @@
+    },
+    onError: (e) => {
+      setSimulateError(e.message);
+      setEstimatedFee(undefined);
+    },
+  });
+  const minDeposit = govParams?.depositParams.min_deposit;
+  const {
+    variant,
+    description: alertDesc,
+    icon,
+  } = getAlert(
+    initialDeposit.amount,
+    minDeposit?.formattedAmount,
+    minDeposit?.formattedDenom
+  );
+
+  const proceed = useCallback(async () => {
+    const minDepositAmount = big(minDeposit?.formattedAmount || 0);
+    const stream = await submitProposalTx({
+      estimatedFee,
+      messages: [submitWhitelistProposalMsg],
+      whitelistNumber: addressesArray.length,
+      amountToVote: minDepositAmount.lte(initialDeposit.amount)
+        ? null
+        : // TODO: Refactor this logic into utils?
+          `${d2Formatter(
+            minDepositAmount.minus(initialDeposit.amount),
+            "NaN"
+          )} ${minDeposit?.formattedDenom}`,
+      onTxSucceed: () => setProcessing(false),
+      onTxFailed: () => setProcessing(false),
+    });
+    if (stream) {
+      setProcessing(true);
+      broadcast(stream);
+    }
+  }, [
+    addressesArray.length,
+    submitWhitelistProposalMsg,
+    estimatedFee,
+    initialDeposit.amount,
+    minDeposit,
+    broadcast,
+    submitProposalTx,
+  ]);
+
+  useEffect(() => {
+    if (router.isReady) AmpTrack(AmpEvent.TO_PROPOSALS);
+  }, [router.isReady]);
+
+  useEffect(() => {
+    if (minDeposit)
+      reset({
+        ...defaultValues,
+        initialDeposit: { denom: minDeposit.denom, amount: "" },
+      });
+  }, [minDeposit, reset]);
+
+  return (
+    <>
+      <PageContainer>
+        <Grid
+          templateAreas={`"prespace main sidebar postspace"`}
+          templateColumns="1fr 6fr 4fr 1fr"
+        >
+          <GridItem area="prespace" />
+          <GridItem area="main">
+            <Heading as="h5" variant="h5">
+              Create Proposal to Whitelisting
+            </Heading>
+            <Text color="text.dark" mt={4} fontWeight={500} variant="body2">
+              Allowed address will be able to upload and stored code without
+              opening proposal
+            </Text>
+            <ConnectWalletAlert
+              subtitle="You need to connect wallet to proceed this action"
+              mt={12}
+            />
+            <form>
+              <Flex
+                mt={12}
+                p={4}
+                gap={6}
+                bgColor="pebble.900"
+                flexDirection="column"
+                borderRadius="8px"
+              >
+                <Flex gap={2} alignItems="center">
+                  <CustomIcon name="proposal-solid" />
+                  <Heading as="h6" variant="h6">
+                    Fill in Proposal Details
+                  </Heading>
+                </Flex>
+                <ControllerInput
+                  name="title"
+                  control={control}
+                  placeholder="ex. Store code for ..."
```

Comment:
should we change placeholder since this is not a proposal to store code?

Reply:
will inquire design team for wording